source/codegen/templates/service_helpers.mako

```diff
@@ -341,12 +341,12 @@ ${populate_response(function_data=function_data, parameters=parameters)}\
   streaming_param_to_include = common_helpers.get_input_streaming_param(parameters)
 %>\
 ${initialize_streaming_input_param(function_name, streaming_param_to_include, parameters, streaming_param)}
-      auto data = std::make_unique<${struct_name}>();\
+      ${struct_name}* data = new ${struct_name}();\
       ${initialize_begin_input_param(streaming_param_to_include, streaming_param)}\
       data->library = std::shared_ptr<${service_class_prefix}LibraryInterface>(library_);
       ${initialize_service_output_params(output_params)}\
       auto moniker = std::make_unique<ni::data_monikers::Moniker>();
-      ni::data_monikers::DataMonikerService::RegisterMonikerInstance("${moniker_function_name}", data.get(), *moniker);
+      ni::data_monikers::DataMonikerService::RegisterMonikerInstance("${moniker_function_name}", data, *moniker);
```

Comment:
If any of those other calls between calling new and what I assume is the RegisterMonikerInstance taking some ownership of the pointer throws an exception. It isn't caught here at least but is higher in the stack we then have a memory leak.

Continue to use a unique_ptr but to show ownership transfer of the pointer use release() instead of get().

Reply:
Yes, tested it out. this works. Resolving this comment.